cranelift-wasm/src/environ/spec.rs

```diff
@@ -378,4 +379,28 @@ pub trait ModuleEnvironment<'data> {
         offset: usize,
         data: &'data [u8],
     );
+
+    /// Declares the name of a module to the environment.
+    ///
+    /// By default this does nothing, but implementations can use this to read
+    /// the module name subsection of the custom name section if desired.
+    fn declare_module_name(&self, _name: &'data str) {}
```

Comment:
Added a method for declaring the module name here, and a method for declaring local names below.